### DISCUSSION TITLE
Riverpod is not updating UI widget on state change

### DISCUSSION THREAD

**Discussion by chhinsras** [rrousselGit/riverpod#1392] [Q&A] [answered]
main.dart

    void main() {
      runApp(ProviderScope(child: MyApp()));
    }
    
    class MyApp extends ConsumerWidget {
      MyApp({Key? key}) : super(key: key);
      final _appProvider =
          ChangeNotifierProvider<AppProvider>((ref) => AppProvider());
    
      final _profileProvider = ChangeNotifierProvider<ProfileProvider>((ref) {
        return ProfileProvider();
      });
    
      @override
      Widget build(BuildContext context, WidgetRef ref) {
        AppProvider appProvider = ref.watch(_appProvider);
        ProfileProvider profileManager = ref.watch(_profileProvider);
        print(appProvider.appLocale);
    }

appProvider.dart

    class AppProvider extends ChangeNotifier {
      Locale? _appLocale = const Locale('en');
    
      Locale get appLocale => _appLocale ?? const Locale("en");
    
      static const supportedLanguage = [Locale('en', 'US'), Locale('km', 'KH')];
        
          void changeLanguage(Locale type) async {
            _appLocale = type;
 
       notifyListeners();
      }
    }
   
[![enter image description here][1]][1]

// when i log value of _appLocale here , I can see the update value based on the state change. but it does not re render UI widgeet.

  [1]: https://i.stack.imgur.com/ubOtQ.png

**Comment by rrousselGit** [comment, marked as answer]
Providers must be globals. You cannot have `_profileProvider` / `__appProvider`  be a property of your widget like that.

**Reply by chhinsras** [reply]
> Providers must be globals. You cannot have `_profileProvider` / `__appProvider` be a property of your widget like that.


how should it be refactored?


 I try to move them out of the class. still not working


```
final _appProvider =
    ChangeNotifierProvider<AppProvider>((ref) => AppProvider());
final _profileProvider = ChangeNotifierProvider<ProfileProvider>((ref) {
  return ProfileProvider();
});
void main() {
  runApp(const ProviderScope(child: MyApp()));
}

class MyApp extends ConsumerWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    AppProvider appProvider = ref.watch(_appProvider);
------
}
}
```

**Comment by chhinsras** [comment]
I fixed it by declaring global provider, remove any deplicated provider, same name, in other files.
